### PR TITLE
[#78] alarm message

### PR DESCRIPTION
**Pull request by JinDDung2**
## 작업 요약

- 메세지 큐 적용

## 작업 내용

> (체크 박스를 모두 선택해야 merge가 활성화 됩니다.)

- [x]  카프카 producer, consumer 
- [x]  Notification 테이블 설계 및 생성
- [x]  상품이 거래되거나 구매자가 구매확정 할 때 판매자에게 알림 전송

### CODE REVIEW

**Comment by JinDDung2** [comment]
### 📝 테스트 커버리지 리포트입니다
|File|Coverage [96.7%]|:green_apple:|
|:-|:-:|:-:|
|[Notification.java](https://github.com/f-lab-edu/givemeticon/blob/a9b44f791da230ce8b029e3616d42008b462c21e/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fnotification%2Fdomain%2FNotification.java)|100%|:green_apple:|
|[NotificationProducer.java](https://github.com/f-lab-edu/givemeticon/blob/a9b44f791da230ce8b029e3616d42008b462c21e/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fnotification%2Fproducer%2FNotificationProducer.java)|100%|:green_apple:|
|[TradeSaleItemUserFacade.java](https://github.com/f-lab-edu/givemeticon/blob/a9b44f791da230ce8b029e3616d42008b462c21e/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Ftrade%2Ffacade%2FTradeSaleItemUserFacade.java)|98.24%|:green_apple:|
|[NotificationConsumer.java](https://github.com/f-lab-edu/givemeticon/blob/a9b44f791da230ce8b029e3616d42008b462c21e/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fnotification%2Fconsumer%2FNotificationConsumer.java)|83.78%|:green_apple:|

|Total Project Coverage|94.39%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/jinddung2/givemeticon/domain/trade/facade/TradeSaleItemUserFacade.java [hunk 1]
```diff
@@ -48,6 +51,8 @@ public int transact(int saleId, int buyerId) {
 
         sale.updateBoughtState();
         saleService.update(sale);
+        producer.create(new CreateNotificationRequestDto(saleId, sale.getSellerId(),
+                String.format("%s이(가) 판매되었습니다.", item.getName())));
```
Comment (f-lab-kyle) [inline src/main/java/com/jinddung2/givemeticon/domain/trade/facade/TradeSaleItemUserFacade.java:55]: 메세지를 만드는 별도의 클래스로 책임을 분리해도 좋을것 같아요.

**Review by f-lab-kyle — COMMENTED** [review]
카프카 까지 도입하셨네요 👍

File: src/main/java/com/jinddung2/givemeticon/domain/trade/facade/TradeSaleItemUserFacade.java (hunk 1 above)
Comment (JinDDung2) [reply src/main/java/com/jinddung2/givemeticon/domain/trade/facade/TradeSaleItemUserFacade.java:55]: 메세지를 만드는 클래스는 유틸리티 클래스처럼 사용하면 될까요?

**Pull request merged**